UnityProject/Assets/Scripts/Items/Weapons/Explosive.cs

```diff
@@ -85,17 +88,32 @@ private void Detonate()
 			if (isServer)
 			{
 				// Get data before despawning
-				var explosionMatrix = registerItem.Matrix;
 				var worldPos = objectBehaviour.AssumedWorldPositionServer();
 
 				// Despawn the explosive
 				_ = Despawn.ServerSingle(gameObject);
+				Explosion.StartExplosion(worldPos, explosiveStrength);
+			}
+
+		}
 
-				// Explosion here
-				var explosionGO = Instantiate(explosionPrefab, explosionMatrix.transform);
-				explosionGO.transform.position = worldPos;
-				explosionGO.Explode(explosionMatrix);
+		[Server]
+		private void CmdAttachExplosive(GameObject target, Vector2 targetPostion)
+		{
+			if (target.TryGetComponent<PushPull>(out var handler))
+			{
+				Inventory.ServerDrop(pickupable.ItemSlot);
+
+				transform.SetParent(handler.transform);
```

Comment:
Dont do this as its not networked, and we shouldnt be parenting networked object.

Luckily i recently found a solution, Constraints!

https://docs.unity3d.com/Manual/Constraints.html

Make a new NetworkedConstraints script, have a sync var of gameobject and then in the hook do the constraint stuff

Reply:
Actually thinking about it, this is gonna be more tricky as the tilemap position of the object isnt moved. 

I think theres an event you can subscribe to the attached objects custom transform or register tile for when it is moved.

And then have the C4 move when that event is invoked.

Don't need to use the constraints/transform this way.

Reply:
I can't find the event that allows this

Reply:
there is one on custom net transform but it doesn't accept functions